pkg/api/serialization_test.go

```diff
@@ -105,9 +105,16 @@ func fuzzInternalObject(t *testing.T, forVersion string, item runtime.Object, se
 		},
 		func(j *deploy.DeploymentStrategy, c fuzz.Continue) {
 			c.FuzzNoCustom(j)
+			mkintp := func(i int) *int64 {
+				v := int64(i)
+				return &v
+			}
 			// TODO: we should not have to set defaults, instead we should be able to detect defaults were applied.
-			if len(j.Type) == 0 {
-				j.Type = deploy.DeploymentStrategyTypeRecreate
+			j.Type = deploy.DeploymentStrategyTypeRolling
```

Comment:
It would be better to select from a number of strategies and ensure they're all round tripped.  As is, there's no longer a guarantee that all of our strategies round trip.

Do something like:

```
switch c.Intn(3) {
case 0:
  // rolling deployment
case 1:
  // recreate
case 2:
  // other
}